wgpu-types/src/lib.rs

```diff
@@ -1087,6 +1087,13 @@ bitflags::bitflags! {
         ///
         /// WebGL doesn't support this.
         const BUFFER_BINDINGS_NOT_16_BYTE_ALIGNED = 1 << 15;
+
+        /// Supports buffers to combine [`BufferUsages::INDEX`] with usages other than [`BufferUsages::COPY_DST`] and [`BufferUsages::COPY_SRC`].
+        /// Furthermore, in absence of this feature it is not allowed to copy index buffers from/to buffers with usages other than
+        /// [`BufferUsages::INDEX`]/[`BufferUsages::COPY_DST`]/[`BufferUsages::COPY_SRC`].
```

Comment:
This needs to include MAP_* as it seems like you can't copy to and from mapped buffers.

Reply:
the doc here is correct then - it says the only things *allowed* to copy from/to are INDEX/COPY_DST/COPY_SRC. But the code in transfer.rs isn't correct and doesn't mind the MAP_WRITE/READ currently.

Reply:
oh, or are you saying there is the additional restriction that index buffers can't be mapped?

Reply:
I phrased this poorly - it's _valid_ to copy to and from mapped buffers (and this is a very common operation) but the docs make it seem like it's invalid.